Add documentation guidelines (#4893)
woocommerce/woocommerce-gutenberg-products-block: docs/contributors/documentation-guidelines.md

  - [Use Automattic Writing Style Guide](#use-automattic-writing-style-guide)
  - [Use active instead of passive voice](#use-active-instead-of-passive-voice)
  - [Use the personal pronoun “you”](#use-the-personal-pronoun-you)
  - [Don’t use gendered pronouns](#dont-use-gendered-pronouns)
  - [Filename must resemble the title](#filename-must-resemble-the-title)
  - [Use correct heading hierarchy](#use-correct-heading-hierarchy)
  - [Use semantically correct markup](#use-semantically-correct-markup)
  - [Use correct spelling](#use-correct-spelling)
  - [Use images not wider than 50% width](#use-images-not-wider-than-50-width)
  - [Use descriptive links](#use-descriptive-links)
  - [Explain arguments](#explain-arguments)
  - [Explicitly define language for code examples](#explicitly-define-language-for-code-examples)
  - [Use table of contents for top-level readme](#use-table-of-contents-for-top-level-readme)
  - [Use internal links](#use-internal-links)
  - [Sort releases descending](#sort-releases-descending)
  - [Structure “How to” instructions](#structure-how-to-instructions)
  - [Link references](#link-references)

## Use Automattic Writing Style Guide

> The Automattic Writing Style Guide serves as the foundation of these documentation guidelines.

Reviewer: If we can add one, a link would be good here :)

Author: Absolutely. I must have missed this.

## Use active instead of passive voice

Active voice should be preferred over passive voice. Passive voice can be used, when it fits better.

**Example**

-   **Active voice:** You want to render your own components in specific places in the Cart and Checkout.
-   **Passive voice:** Slots and Fills add the possibility to render your own HTML in pre-defined places in the Cart and Checkout.

## Use the personal pronoun “you”

When using active voice, the second person singular (you) must be used.

## Don’t use gendered pronouns

Gendered pronouns (she/her/hers and he/him/his) must not be used.

## Filename must resemble the title

The filename of the document must match the title.

**Example**

-   **Title:** Slots and Fills
-   **Filename:** slots-and-fills.md

## Use correct heading hierarchy

> Correct H1-H6 headings must be used. Each document can only have one H1 heading. An H3 heading can only be used within an H2 heading, an H4 heading can only be used within an H3 heading, etc. Headings must not use articles.

Reviewer: > Headings must not use articles

Perhaps we could word it more like "If possible, avoid using articles in headings" sometimes it might be necessary.

Author: That makes sense. Thanks for this suggestion.

**Example**

-   **Incorrect:** The problem
-   **Correct:** Problem

## Use semantically correct markup

> The markup must be used semantically correct, e.g. a list items must be formatted as list.

Reviewer: ```suggestion
The markup used must be semantically correct, e.g. list markup must only be used to display a list.
```
